src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,297 @@
+     *
+     * @param entity The entity to check
+     * @return Boolean value of whether the entity is a barbarian
+     */
+    public static Boolean isBarbarian(final Entity entity)
+    {
+        return (entity instanceof EntityBarbarian || entity instanceof EntityArcherBarbarian || entity instanceof EntityChiefBarbarian);
+    }
+
+    /**
+     * Returns the closest barbarian to an entity.
+     *
+     * @param entity             The entity to test against
+     * @param distanceFromEntity The distance to check for
+     * @return the barbarian (if any) that is nearest
+     */
+    public static Entity getClosestBarbarianToEntity(final Entity entity, final double distanceFromEntity)
+    {
+        final List<Entity> entityList = CompatibilityUtils.getWorld(entity).getEntitiesInAABBexcluding(
+          entity,
+          entity.getEntityBoundingBox().expand(
+            distanceFromEntity,
+            3.0D,
+            distanceFromEntity),
+          Entity::isEntityAlive);
+
+        final Optional<Entity> entityBarbarian = entityList.stream()
+                                                   .filter(BarbarianUtils::isBarbarian)
+                                                   .findFirst();
+
+        return entityBarbarian.orElse(null);
+    }
+
+    /**
+     * Returns the barbarians close to an entity.
+     *
+     * @param entity             The entity to test against
+     * @param distanceFromEntity The distance to check for
+     * @return the barbarians (if any) that is nearest
+     */
+    public static Stream<EntityLivingBase> getBarbariansCloseToEntity(final Entity entity, final double distanceFromEntity)
+    {
+        final List<EntityLivingBase> entityList = CompatibilityUtils.getWorld(entity).getEntitiesWithinAABB(
+          EntityLivingBase.class,
+          entity.getEntityBoundingBox().expand(
+            distanceFromEntity,
+            3.0D,
+            distanceFromEntity),
+          Entity::isEntityAlive);
+
+        return entityList.stream().filter(BarbarianUtils::isBarbarian);
+    }
+
+    /**
+     * Enacts the Raid Event.
+     *
+     * @param colony The colony for which the raid is occurring in.
+     */
+    public static void EventRaid(final Colony colony)
+    {
+        final World world = colony.getWorld();
+
+        if (world == null)
+        {
+            return;
+        }
+
+        final boolean isDay = world.isDaytime();
+
+        if (isDay && colony.getHasRaided())
+        {
+            colony.setRaidWillHappen(raidThisNight(world));
+            colony.setHasRaided(false);
+        }
+        else if (isDay && !colony.getHasRaided())
+        {
+            colony.setHasRaided(false);
+        }
+        else if (!isDay && !colony.getHasRaided() && colony.getWillRaid() && Configurations.doBarbariansSpawn && (world.getDifficulty() != EnumDifficulty.PEACEFUL))
+        {
+            System.out.println("Did the thing");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS106)